xgi/utils/utilities.py

```diff
@@ -73,3 +74,51 @@ def load_xgi_data(dataset, nodetype=None, edgetype=None):
     r = requests.get(index[dataset]["url"])
 
     return convert.dict_to_hypergraph(r.json(), nodetype=nodetype, edgetype=edgetype)
+
+
+def convert_labels_to_integers(H):
+    """Relabel node and edge IDs to be sequential integers.
+
+    Parameters
+    ----------
+    H : Hypergraph
+        The hypergraph of interest
+
+    Returns
+    -------
+    Hypergraph
+        A new hypergraph with nodes and edges with sequential IDs starting at 0.
+        The old IDs are stored in the "label" attribute for both nodes and edges.
+
+    Notes
+    -----
+    The "relabeling" will occur even if the node/edge IDs are sequential.
+    Because the old IDs are stored in the "label" attribute for both nodes and edges,
+    the old "label" values (if they exist) will be overwritten.
+    """
+    node_id = count()
+    edge_id = count()
+    node_mapping = dict()
+    edge_mapping = dict()
+    temp_H = H.copy()
+    for node in list(temp_H.nodes):
+        id = next(node_id)
+        temp_H._node[id] = temp_H._node.pop(node)
```

Comment:
I'm pretty sure this is buggy. For example, if the original hypergraph has a node with id `0` (but other nodes have ids that are, say, strings), the first time this line is executed, the edges incident to `0` will be overwritten and can never be recovered.

My suggestion is to first build the `node_mapping` and `edge_mapping` dictionaries WITHOUT modifying the network, and then using those mappings to do the relabeling in a single loop. In this loop, you are trying to build the `node_mapping` at the same time as modifying the network, and that's what's causing the problem.

Reply:
This is a good suggestion. Done.